### PR TITLE
Deprecate `@itwin/core-react`

### LINKED ISSUE #827: Deprecate `@itwin/core-react` package
### Describe the Solution you'd like

Most of `@itwin/core-react` was already replaced by `iTwinUI`. We should deprecate the remaining components and APIs from this package in preparation of package removal. In case the API is still needed and there's no available replacement - move it to `@itwin/components-react`.

```[tasklist]
### Tasks
- [x] Deprecate `Popup` in favor of `iTwinUI` `Popover`
- [ ] Move remaining components to other packages
```

### PR DESCRIPTION
**Pull request by GerardasB**
## Changes

This PR closes #827 by deprecating the remaining APIs from `@itwin/core-react` package. Most of the components are accessible from [iTwinUI](https://itwinui.bentley.com), some common types used through other AppUI packages were moved to either `@itwin/components-react` or `@itwin/appui-react`. Additionally, `IconSpec` deprecation led to a number of property deprecations in `@itwin/appui-react`.
See `NextVersion.md` for a full list of deprecations and their suggested replacements.

## Testing

Added additional stories.

### CODE REVIEW

**Review by saskliutas — APPROVED** [review]

**Review by ignas-k — COMMENTED** [review]
Nice, left a few nit-picks.

**Review by ignas-k — APPROVED** [review]

File: e2e-tests/tests/backstage/backstage.test.ts-snapshots/backstage-test-with-snapshotPath-1-chromium-linux.png
[the remark this answers is not in the document]
Comment (GerardasB) [reply e2e-tests/tests/backstage/backstage.test.ts-snapshots/backstage-test-with-snapshotPath-1-chromium-linux.png:1]: Good point, fixed

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -164,9 +200,16 @@ Table of contents:
...
-- Added replacement properties that allow specifying the badge type in components using string values from `BadgeKind`. [#899](https://github.com/iTwin/appui/pull/899)
-  - `badgeKind` property in `CommonBackstageItem`, `CommonStatusBarItem`, `CommonToolbarItem`, `Widget`, `CursorMenuItemProps`, `FrameworkKeyboardShortcut`, `ItemDefBase`, `ItemProps`, `KeyboardShortcutProps`.
-  - `badgeKind` getter in `WidgetDef`.
+  - Added replacement properties that allow specifying the badge type in components using string values from `BadgeKind`. [#899](https://github.com/iTwin/appui/pull/899)
+    - `badgeKind` property in `CommonBackstageItem`, `CommonStatusBarItem`, `CommonToolbarItem`, `Widget`, `CursorMenuItemProps`, `FrameworkKeyboardShortcut`, `ItemDefBase`, `ItemProps`, `KeyboardShortcutProps`.
+    - `badgeKind` getter in `WidgetDef`.
+  - Added `LocalStateStorage`, `SettingsManager` classes, `UiStateStorageStatus` enum and `UiStateStorage`, `UiStateStorageResult`, `RectangleProps`, `SizeProps`, `SettingsTabEntry`, `SettingsTabsProvider` interfaces previously accessible from `@itwin/core-react` package. [#901](https://github.com/iTwin/appui/pull/901)
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply docs/changehistory/NextVersion.md]: Fixed

File: docs/changehistory/NextVersion.md [hunk 2]
```diff
@@ -165,9 +200,16 @@ Table of contents:
   - `onViewSelected` to `ViewSelectorProps` to replace the static event.
   - `solarTimeline`, `analysisTimeline`, `scheduleAnimation` props to `DefaultViewOverlay` component to replace existing `applicationData.featureOptions.defaultViewOverlay` property.
   - `navigationAid` prop to `ViewToolWidgetComposer` to override the default navigation aid.
-- Added replacement properties that allow specifying the badge type in components using string values from `BadgeKind`. [#899](https://github.com/iTwin/appui/pull/899)
-  - `badgeKind` property in `CommonBackstageItem`, `CommonStatusBarItem`, `CommonToolbarItem`, `Widget`, `CursorMenuItemProps`, `FrameworkKeyboardShortcut`, `ItemDefBase`, `ItemProps`, `KeyboardShortcutProps`.
-  - `badgeKind` getter in `WidgetDef`.
+  - Added replacement properties that allow specifying the badge type in components using string values from `BadgeKind`. [#899](https://github.com/iTwin/appui/pull/899)
+    - `badgeKind` property in `CommonBackstageItem`, `CommonStatusBarItem`, `CommonToolbarItem`, `Widget`, `CursorMenuItemProps`, `FrameworkKeyboardShortcut`, `ItemDefBase`, `ItemProps`, `KeyboardShortcutProps`.
+    - `badgeKind` getter in `WidgetDef`.
```
Comment (ignas-k) [inline docs/changehistory/NextVersion.md:205]: I believe this block needs to have one indentation level removed as well.

**Pull request merged**